repository/src/lib.rs

```diff
@@ -168,9 +168,206 @@ impl<T: RawRepository> DistributedRepository<T> {
+                .collect::<Result<Vec<_>, _>>();
+            if new_commits.is_err() {
+                log::warn!("Found invalid remote tracking branches");
+                continue 'branch;
+            }
+            let new_commits = new_commits.unwrap();
+
+            // Verify all the incoming commits and apply them to the local repository only if they are valid.
+            let mut verifier =
+                CommitSequenceVerifier::new(last_header.clone(), reserved_state.clone())
+                    .map_err(|e| anyhow!("failed to create a verifier: {}", e))?;
+            for (new_commit, _) in &new_commits {
+                if verifier.apply_commit(new_commit).is_err() {
+                    continue 'branch;
+                }
+            }
+            let branch_tip_commit = &new_commits.last().unwrap().0;
+            match branch_tip_commit {
+                Commit::Agenda(_) => {
+                    // If branch tip commit is Agenda commit, create a new agenda branch
+                    let branch_tip_commit_hash = self.raw.locate_branch(branch.clone()).await?;
+                    let branch_name = format!(
+                        "a-{:?}",
+                        branch_tip_commit
+                            .to_hash256()
+                            .to_string()
+                            .truncate(COMMIT_TITLE_HASH_DIGITS)
+                    );
+                    self.raw
+                        .create_branch(branch_name, branch_tip_commit_hash)
+                        .await?;
+                }
+                Commit::AgendaProof(_) => {
+                    // Else if branch tip commit is AgendaProof commit, create a new agenda branch
+                    let branch_tip_commit_hash = self.raw.locate_branch(branch.clone()).await?;
+                    let branch_name = format!(
+                        "a-{:?}",
+                        branch_tip_commit
+                            .to_hash256()
+                            .to_string()
+                            .truncate(COMMIT_TITLE_HASH_DIGITS)
+                    );
+                    self.raw
+                        .create_branch(branch_name, branch_tip_commit_hash)
+                        .await?;
+                }
+                Commit::Block(block_header) => {
+                    // Verify finalization proof
+                    let fp_commit_hash = self.raw.locate_branch(FP_BRANCH_NAME.into()).await?;
+                    // TODO: change this FP_BRANCH_NAME to remote one
+                    let fp_semantic_commit = self.raw.read_semantic_commit(fp_commit_hash).await?;
+                    let finalization_proof: FinalizationProof =
+                        serde_json::from_str(&fp_semantic_commit.body).unwrap();
+                    let result =
+                        verify::verify_finalization_proof(block_header, &finalization_proof);
+                    match result {
+                        Ok(()) => {
```

Comment:
Can you add a comment for Ok case as well? (like Err case)

Reply:
Added.